### PR TITLE
Database refractoring

### PR DESCRIPTION
**Pull request by mominul**
* This merges the data entries which are shared by both fixed and phonetic methods.
* So this will reduce the lag while switching between Fixed and Phonetic methods, as they were reloaded again and again previously.
*  Update the phonetic selection entries if only the file was modified in the meantime.

cc @gulshan can you have a look?

### CODE REVIEW

**Comment by gulshan** [comment]
I think, I have to download the pr to review. May be it'll take some time.

File: src/data.rs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    pub(crate) fn get_emoji_by_name(&self, name: &str) -> Option<impl Iterator<Item = &str>> {
+        self.emojicon.get_by_name(name)
+    }
+
+    pub(crate) fn get_emoji_by_bengali(&self, name: &str) -> Option<impl Iterator<Item = &str>> {
+        self.bengali_emoji.get(name)
+    }
+}
+
+mod tests {
```
Comment (gulshan) [inline src/data.rs:56]: Please add `#[cfg(test)]` here to stop the warnings.

Comment (mominul) [reply src/data.rs:56]: Oops, missed that!

**Comment by gulshan** [comment]
I have a query. Is this PR all about refactoring, or it adds features also?

**Comment by mominul** [comment]
> I have a query. Is this PR all about refactoring, or it adds features also?

It doesn't, but adds optimizations like updating the phonetic selection entries if only the file was modified in the meantime and not loading data that are sharable.
But my next work, adding Bijoy2000 support will depend on this refactoring, as the converter instance should be shared between the methods to not load the conversion map every time.

**Comment by mominul** [comment]
@gulshan Thanks for the initial review! Sorry to say, I am short on time, so I'll have this merged so I could do the remaining work. Can you please raise your concerns in an issue or a PR?

**Pull request merged**